ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BVC+WKNavigationDelegate.swift

```diff
@@ -689,13 +692,8 @@ extension BrowserViewController: WKNavigationDelegate {
       let response = response as? HTTPURLResponse
     {
       let internalUrl = InternalURL(responseURL)
-      let isErrorPage = internalUrl?.isErrorPage == true || response.statusCode >= 400
-      let isSessionRestore = internalUrl?.isSessionRestore == true
-
-      if isErrorPage || isSessionRestore {
-        tab.shouldNotifyAdsServiceTabDidChange = false
-        tab.shouldNotifyAdsServiceTabContentDidChange = false
-      }
+      tab.isErrorPage = internalUrl?.isErrorPage == true || response.statusCode >= 400
+      tab.isRestored = tab.isRestored || internalUrl?.isSessionRestore == true
```

Comment:
Out of interest what is the reasoning behind needing both conditions?

Reply:
tab.isRestored is already can be true here, because can be set during tab state restoration: https://github.com/brave/brave-core/blob/7389202cfcb82939bd6e404d94c3e127d5841f1d/ios/brave-ios/Sources/Brave/Frontend/Browser/Tab.swift#L573

so these conditions are similar to if statement:
```
if !tab.isRestored {
  tab.isRestored = internalUrl?.isSessionRestore == true
}
```

I will change it to if statement to make it clearer

Reply:
Fixed